pkg/controller/database/rdsinstance.go

```diff
@@ -142,12 +142,17 @@ func (e *external) createAccountIfNeeded(cr *v1alpha1.RDSInstance) (string, erro
 	err = e.client.CreateAccount(cr.Status.AtProvider.DBInstanceID, cr.Spec.ForProvider.MasterUsername, pw)
 	if err != nil {
 		// The previous request might fail due to timeout. That's fine we will eventually reconcile it.
-		if sdkErr, ok := err.(sdkerror.Error); ok {
-			if sdkErr.ErrorCode() == "InvalidAccountName.Duplicate" {
-				cr.Status.AtProvider.AccountReady = true
-				return "", nil
-			}
+		var sdkerr sdkerror.Error
+
+		if !errors.As(err, sdkerr) {
+			return "", err
+		}
```

Comment:
@jastang I believe this (and the same situation in a few other places) will panic. We'll need to pass `&sdkerr` here to satisfy `errors.As`

Reply:
thank you for finding the misses @hasheddan!